liblocation/src/main/java/com/mapbox/android/core/location/LocationEngineProvider.java

```diff
@@ -67,13 +75,35 @@ private void initAvailableLocationEngines(Context context) {
   }
 
   private LocationEngine obtainBestLocationEngine() {
-    LocationEngine androidLocationEngine = locationEngineDictionary.get(LocationEngine.Type.ANDROID);
-    for (LocationEngine.Type type : OPTIONAL_LOCATION_ENGINES) {
-      LocationEngine bestLocationEngine = locationEngineDictionary.get(type);
-      if (bestLocationEngine != null) {
-        return bestLocationEngine;
-      }
+    LocationEngine bestLocationEngine;
+
+    // Check to see if google play services is installed and updated to required version
+    if (googlePlayAvailabilityResultCode == ConnectionResult.SUCCESS) {
+      bestLocationEngine = getGoogleLocationEngine();
+    } else {
+      bestLocationEngine = getAndroidLocationEngine();
+    }
+    return bestLocationEngine;
+  }
+
+  @Nullable
```

Comment:
Is it really `Nullable`? `obtainBestLocationEngineAvailable ` which calls `obtainBestLocationEngine` then returns `NonNull`

Reply:
It's because of `obtainLocationEngineBy(type)` which is `Nullable`. I think method `getGoogleLocationEngine` should be `Nullable` too, I forgot to annotate it.
But if we are sure that `obtainLocationEngineBy` is NonNull then we can remove `Nullable` annotations.

Reply:
I'd suggest that `LocationEngine.Type.ANDROID` is always present while `LocationEngine.Type.GOOGLE_PLAY_SERVICES` is not since it depends on google services status.

Reply:
Thanks for the comment, I removed it.